// @tikz/hedera-mirror-node-ts/dist/RestMirrorNode/nftBaseClasses/NFTTransactionHistory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NFTTransactionHistory = void 0;
const __1 = require("../");
class NFTTransactionHistory extends __1.HasMoreMirrorNode {
    constructor(mirrorClient, url, tokenId, serialNumber) {
        super(mirrorClient);
        this.url = url;
        this.tokenId = '';
        this.serialNumber = 1;
        this.params = {};
        if (tokenId)
            this.tokenId = tokenId;
        if (serialNumber)
            this.serialNumber = serialNumber;
    }
    static v1(mirrorClient, tokenId, serialNumber) {
        return new this(mirrorClient, `/api/v1/tokens`, tokenId, serialNumber);
    }
    setTokenId(val) {
        this.tokenId = val;
        return this;
    }
    setSerialNumber(val) {
        this.serialNumber = val;
        return this;
    }
    get() {
        if (!this.tokenId || !this.serialNumber)
            throw new Error('token id or serial number not set');
        return this.fetch(`${this.url}/${this.tokenId}/nfts/${this.serialNumber}/transactions`);
    }
}
exports.NFTTransactionHistory = NFTTransactionHistory;
