// @tikz/hedera-mirror-node-ts/dist/RestMirrorNode/helpers/filterKeys.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.filterKeys = void 0;
var filterKeys;
(function (filterKeys) {
    filterKeys["ACCOUNT_ID"] = "account.id";
    filterKeys["ACCOUNT_BALANCE"] = "account.balance";
    filterKeys["ACCOUNT_PUBLICKEY"] = "account.publickey";
    filterKeys["BALANCE"] = "balance";
    filterKeys["CONTRACTID"] = "contractid";
    filterKeys["CONTRACT_ID"] = "contract.id";
    filterKeys["CREDIT_TYPE"] = "type";
    filterKeys["ENCODING"] = "encoding";
    filterKeys["FROM"] = "from";
    filterKeys["ENTITY_PUBLICKEY"] = "publickey";
    filterKeys["INDEX"] = "index";
    filterKeys["LIMIT"] = "limit";
    filterKeys["NONCE"] = "nonce";
    filterKeys["ORDER"] = "order";
    filterKeys["RESULT"] = "result";
    filterKeys["SCHEDULED"] = "scheduled";
    filterKeys["SCHEDULEID"] = "scheduleid";
    filterKeys["SCHEDULE_ID"] = "schedule.id";
    filterKeys["SERIAL_NUMBER"] = "serialnumber";
    filterKeys["SEQUENCE_NUMBER"] = "sequencenumber";
    filterKeys["TIMESTAMP"] = "timestamp";
    filterKeys["TOKENID"] = "tokenid";
    filterKeys["TOKEN_ID"] = "token.id";
    filterKeys["TOKEN_TYPE"] = "type";
    filterKeys["TOPIC_ID"] = "topic.id";
    filterKeys["TRANSACTION_TYPE"] = "transactiontype";
})(filterKeys = exports.filterKeys || (exports.filterKeys = {}));
