// @tikz/hedera-mirror-node-ts/dist/RestMirrorNode/helpers/TransactionType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionType = void 0;
var TransactionType;
(function (TransactionType) {
    TransactionType["CONSENSUSCREATETOPIC"] = "CONSENSUSCREATETOPIC";
    TransactionType["CONSENSUSDELETETOPIC"] = "CONSENSUSDELETETOPIC";
    TransactionType["CONSENSUSSUBMITMESSAGE"] = "CONSENSUSSUBMITMESSAGE";
    TransactionType["CONSENSUSUPDATETOPIC"] = "CONSENSUSUPDATETOPIC";
    TransactionType["CONTRACTCALL"] = "CONTRACTCALL";
    TransactionType["CONTRACTCREATEINSTANCE"] = "CONTRACTCREATEINSTANCE";
    TransactionType["CONTRACTDELETEINSTANCE"] = "CONTRACTDELETEINSTANCE";
    TransactionType["CONTRACTUPDATEINSTANCE"] = "CONTRACTUPDATEINSTANCE";
    TransactionType["CRYPTOADDLIVEHASH"] = "CRYPTOADDLIVEHASH";
    TransactionType["CRYPTOCREATEACCOUNT"] = "CRYPTOCREATEACCOUNT";
    TransactionType["CRYPTODELETE"] = "CRYPTODELETE";
    TransactionType["CRYPTODELETELIVEHASH"] = "CRYPTODELETELIVEHASH";
    TransactionType["CRYPTOTRANSFER"] = "CRYPTOTRANSFER";
    TransactionType["CRYPTOUPDATEACCOUNT"] = "CRYPTOUPDATEACCOUNT";
    TransactionType["FILEAPPEND"] = "FILEAPPEND";
    TransactionType["FILECREATE"] = "FILECREATE";
    TransactionType["FILEDELETE"] = "FILEDELETE";
    TransactionType["FILEUPDATE"] = "FILEUPDATE";
    TransactionType["FREEZE"] = "FREEZE";
    TransactionType["SCHEDULECREATE"] = "SCHEDULECREATE";
    TransactionType["SCHEDULEDELETE"] = "SCHEDULEDELETE";
    TransactionType["SCHEDULESIGN"] = "SCHEDULESIGN";
    TransactionType["SYSTEMDELETE"] = "SYSTEMDELETE";
    TransactionType["SYSTEMUNDELETE"] = "SYSTEMUNDELETE";
    TransactionType["TOKENASSOCIATE"] = "TOKENASSOCIATE";
    TransactionType["TOKENBURN"] = "TOKENBURN";
    TransactionType["TOKENCREATION"] = "TOKENCREATION";
    TransactionType["TOKENDELETION"] = "TOKENDELETION";
    TransactionType["TOKENDISSOCIATE"] = "TOKENDISSOCIATE";
    TransactionType["TOKENFEESCHEDULEUPDATE"] = "TOKENFEESCHEDULEUPDATE";
    TransactionType["TOKENFREEZE"] = "TOKENFREEZE";
    TransactionType["TOKENGRANTKYC"] = "TOKENGRANTKYC";
    TransactionType["TOKENMINT"] = "TOKENMINT";
    TransactionType["TOKENPAUSE"] = "TOKENPAUSE";
    TransactionType["TOKENREVOKEKYC"] = "TOKENREVOKEKYC";
    TransactionType["TOKENUNFREEZE"] = "TOKENUNFREEZE";
    TransactionType["TOKENUNPAUSE"] = "TOKENUNPAUSE";
    TransactionType["TOKENUPDATE"] = "TOKENUPDATE";
    TransactionType["TOKENWIPE"] = "TOKENWIPE";
    TransactionType["UNCHECKEDSUBMIT"] = "UNCHECKEDSUBMIT";
})(TransactionType = exports.TransactionType || (exports.TransactionType = {}));
