// @tikz/hedera-mirror-node-ts/dist/RestMirrorNode/tokenBaseClasses/TokenBalances.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenBalances = void 0;
const __1 = require("..");
class TokenBalances extends __1.HasMoreMirrorNode {
    constructor(mirrorClient, url, tokenId) {
        super(mirrorClient);
        this.url = url;
        this.params = {};
        this.tokenId = '';
        if (tokenId)
            this.tokenId = tokenId;
    }
    static v1(mirrorClient, tokenId) {
        return new this(mirrorClient, `/api/v1/tokens`, tokenId);
    }
    setPublicKey(val) {
        this.params[__1.filterKeys.ACCOUNT_PUBLICKEY] = val;
        return this;
    }
    setAccountBalance(val) {
        this.params[__1.filterKeys.ACCOUNT_BALANCE] = val;
        return this;
    }
    setAccountId(val) {
        this.params[__1.filterKeys.ACCOUNT_ID] = val;
        return this;
    }
    setTokenId(val) {
        this.tokenId = val;
        return this;
    }
    get() {
        if (!this.tokenId)
            throw new Error('no token id');
        return this.fetch(`${this.url}/${this.tokenId}/balances`);
    }
}
exports.TokenBalances = TokenBalances;
