// @tikz/hedera-mirror-node-ts/dist/RestMirrorNode/Accounts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Accounts = void 0;
const _1 = require("./");
const {} = _1.TransactionType;
class Accounts extends _1.HasMoreMirrorNode {
    constructor(mirrorNodeClient, url, accountId) {
        super(mirrorNodeClient);
        this.url = url;
        this.params = {};
        if (accountId)
            this.setAccountId(accountId);
    }
    /**
     * returns Accounts Mirror Client with version 1
     */
    static v1(mirrorNodeClient, accountId) {
        return new this(mirrorNodeClient, "/api/v1/accounts", accountId);
    }
    setBalance(val) {
        this.params[_1.filterKeys.ACCOUNT_BALANCE] = val;
        return this;
    }
    setAccountId(val) {
        this.params[_1.filterKeys.ACCOUNT_ID] = val;
        return this;
    }
    setPublicKey(val) {
        this.params[_1.filterKeys.ACCOUNT_PUBLICKEY] = val;
        return this;
    }
    setTransactionType(val) {
        this.params[_1.filterKeys.TRANSACTION_TYPE] = val;
        return this;
    }
    async get() {
        return this.fetch(this.url);
    }
}
exports.Accounts = Accounts;
