// @tikz/hedera-mirror-node-ts/dist/RestMirrorNode/Transactions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transactions = void 0;
const _1 = require("./");
class Transactions extends _1.HasMoreMirrorNode {
    constructor(mirrorNodeClient, url, accountId) {
        super(mirrorNodeClient);
        this.url = url;
        this.params = {};
        if (accountId)
            this.setAccountId(accountId);
    }
    static v1(mirrorNodeClient, accountId) {
        return new this(mirrorNodeClient, '/api/v1/transactions', accountId);
    }
    setAccountId(val) {
        this.params[_1.filterKeys.ACCOUNT_ID] = val;
        return this;
    }
    setResult(val) {
        this.params[_1.filterKeys.RESULT] = val;
        return this;
    }
    setType(val) {
        this.params[_1.filterKeys.CREDIT_TYPE] = val;
        return this;
    }
    setTransactionType(val) {
        this.params[_1.filterKeys.TRANSACTION_TYPE] = val;
        return this;
    }
    async get() {
        return this.fetch(this.url);
    }
}
exports.Transactions = Transactions;
