// @tikz/hedera-mirror-node-ts/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.smartContract = exports.nftUtils = exports.tokenUtils = exports.transactions = exports.networkSupply = exports.topicMessages = exports.accounts = void 0;
__exportStar(require("./RestMirrorNode"), exports);
__exportStar(require("./AxiosMirrorNodeClient"), exports);
const _1 = require("./");
const RestMirrorNode_1 = require("./RestMirrorNode");
const accounts = (client) => _1.Accounts.v1(client);
exports.accounts = accounts;
const topicMessages = (client) => _1.TopicMessages.v1(client);
exports.topicMessages = topicMessages;
const networkSupply = (client) => _1.NetworkSupply.v1(client);
exports.networkSupply = networkSupply;
const transactions = (client) => _1.Transactions.v1(client);
exports.transactions = transactions;
const tokenUtils = (client) => RestMirrorNode_1.TokenUtils.v1(client);
exports.tokenUtils = tokenUtils;
const nftUtils = (client) => RestMirrorNode_1.NFTUtils.v1(client);
exports.nftUtils = nftUtils;
const smartContract = (client) => RestMirrorNode_1.SmartContracts.v1(client);
exports.smartContract = smartContract;
